sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -540,20 +540,26 @@ object QueryExecution {
   }
 
   /**
-   * Converts asserts, null pointer exceptions to internal errors.
+   * Converts asserts, null pointer exceptions and scala match errors to internal errors.
    */
   private[sql] def toInternalError(msg: String, e: Throwable): Throwable = e match {
     case e @ (_: java.lang.NullPointerException | _: java.lang.AssertionError) =>
```

Comment:
Why don't you add just one more case of `scala.MatchError` here?

The devs who will investigate the issue will see the cause exception `scala.MatchError`.

Reply:
Merged `scala.MatchError` with other two exceptions.